libcxx/test/std/utilities/expected/expected.expected/observers/has_value_clobber.pass.cpp

```diff
@@ -0,0 +1,36 @@
+//===----------------------------------------------------------------------===//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
+// ADDITIONAL_COMPILE_FLAGS: -O2
```

Comment:
I needed to compile with optimizations to make this test fail without the patch.

Reply:
FWIW I am actually getting the failure even without optimizations. I am using AppleClang.